programs/bpf_loader/src/syscalls.rs

```diff
@@ -2242,7 +2242,8 @@ fn check_authorized_program(
         || bpf_loader_deprecated::check_id(program_id)
         || (bpf_loader_upgradeable::check_id(program_id)
             && !(bpf_loader_upgradeable::is_upgrade_instruction(instruction_data)
-                || bpf_loader_upgradeable::is_set_authority_instruction(instruction_data)))
+                || bpf_loader_upgradeable::is_set_authority_instruction(instruction_data)
+                || bpf_loader_upgradeable::is_close_instruction(instruction_data)))
```

Comment:
need a feature switch here because closing a buffer could succeed before the `close_upgradeable_program_accounts` feature switch is enabled